x-pack/plugins/security_solution/common/entity_analytics/asset_criticality/parse_asset_criticality_csv_row.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { CriticalityLevels } from './constants';
+import { ValidCriticalityLevels } from './constants';
+import type { AssetCriticalityUpsert, CriticalityLevel } from './types';
+
+const MAX_COLUMN_CHARS = 1000;
+
+interface ValidRecord {
+  valid: true;
+  record: AssetCriticalityUpsert;
+}
+interface InvalidRecord {
+  valid: false;
+  error: string;
+}
+
+type ReturnType = ValidRecord | InvalidRecord;
+
+export const isErrorResult = (result: ReturnType): result is InvalidRecord => {
+  return !result.valid;
+};
```

Comment:
Nitpick:

This is essentially the `Either` ADT, do I'd rather just use that instead of implementing it again. That way we get all those type predicates and utility functions for free if we ever need them

Reply:
I have had a go but it just seems more complicated to me, admittedly I am not well versed in fp-ts, can discuss this in the meeting too

Reply:
We decided to leave it as it is in the review meeting